build.tcl

```diff
@@ -378,31 +391,37 @@ proc _install {args} {
 		set version {}
 	    }
 
+	    set namevers $dstl/[pkgdirname $name $version]
+
 	    file copy   -force $selfdir/lib/$dir     $dstl/${name}-new
-	    file delete -force $dstl/$name$version
-	    file rename        $dstl/${name}-new     $dstl/$name$version
-	    puts "${prefix}Installed package:      $dstl/$name$version"
+	    file delete -force $namevers
+	    puts "${prefix}Installed package:      $namevers"
+	    file rename        $dstl/${name}-new     $namevers
 	    set prefix {}
 	}
 
+
 	# Application: critcl
 
 	set theapp  [critapp     $dsta]
 	set reldstl [dstlfromlib $dstl]
 
 	set c [open $theapp w]
 	lappend map @bs@   "\\"
-	lappend map @exe@  [file dirname [file normalize [info nameofexecutable]/___]]
-	lappend map @path@ $reldstl  ;# insert the dst path
+	lappend map @exe@ [shquote [file dirname [file normalize [info nameofexecutable]/...]]]
+	lappend map @path@ [list $reldstl]  ;# insert the dst path
 	lappend map "\t    " {} ;# de-dent
 	puts $c [string trimleft [string map $map {
 	    #!/bin/sh
-	    # -*- tcl -*- @bs@
-	    exec "@exe@" "$0" ${1+"$@"}
-
-	    set libpath [file normalize [file join [file dirname [info script]] .. lib]]
-	    set libpath [file join $libpath {@path@}]
-	    if {[lsearch -exact $auto_path $libpath] < 0} {lappend auto_path $libpath}
+	    # -*- hide next line from tcl -*- @bs@
```

Comment:
Would IMHO be better as `hide next line from -*- tcl -*-`. The `-*- tcl -*-` is an emacs marker to signal the major mode. Would like to keep that signal.